libbeat/common/transport/tlscommon/verify.go

```diff
@@ -62,16 +63,9 @@ func verifyCertificateExceptServerName(
 	rawCerts [][]byte,
 	c *TLSConfig,
 ) ([]*x509.Certificate, [][]*x509.Certificate, error) {
-	// this is where we're a bit suboptimal, as we have to re-parse the certificates that have been presented
-	// during the handshake.
-	// the verification code here is taken from verifyServerCertificate in crypto/tls/handshake_client.go:824
-	certs := make([]*x509.Certificate, len(rawCerts))
-	for i, asn1Data := range rawCerts {
-		cert, err := x509.ParseCertificate(asn1Data)
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "tls: failed to parse certificate from server")
-		}
-		certs[i] = cert
+	certs, err := overwriteSANWithCommonName(rawCerts)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to overwrite SAN with CommonName")
```

Comment:
This method excludes the hostname check, do we really need to ensure "CommonName" working here?